### PR TITLE
DEV2-3003 align status bar notifications

### PR DESCRIPTION
**Pull request by avnerbarr**

### CODE REVIEW

**Review by yanir-codota — COMMENTED** [review]
LGTM. 
would love if @dimacodota would take a look too.

File: src/statusBar/pollStatusBar.ts [hunk 1]
```diff
@@ -3,21 +3,25 @@ import { getStatus } from "../binary/requests/statusBar";
...
   disposeStatusBarCommand,
 } from "./statusBarActionHandler";
+import { statePoller } from "../state/statePoller";
 
 let statusPollingInterval: NodeJS.Timeout | null = null;
 
 export default function pollStatuses(
   context: vscode.ExtensionContext
 ): vscode.Disposable {
+  const statePollerDisposable = statePoller.event((change) => {
```
Comment (dimacodota) [inline src/statusBar/pollStatusBar.ts:19]: 1 - what problem does this solves?
2 - why do we need the both pollings here  ? ie statePoller and doPollStatus ?

File: src/registration/forceRegistration.ts [hunk 2]
```diff
@@ -122,22 +124,35 @@ async function notifyState() {
   }
 }
 
+let awaiting = false;
 function awaitLoginNotification() {
+  if (awaiting) {
```
Comment (dimacodota) [inline src/registration/forceRegistration.ts:129]: remove all the awaiting, and just return a promise which is resolved after the notifications is shown, promise is resolved only once, but you need to keep the same instance of it

File: src/statusBar/pollStatusBar.ts (hunk 1 above)
Comment (avnerbarr) [reply src/statusBar/pollStatusBar.ts:19]: they are emitting at different intervals

File: src/registration/forceRegistration.ts (hunk 2 above)
Comment (avnerbarr) [reply src/registration/forceRegistration.ts:129]: why is that easiser / better?

File: src/statusBar/pollStatusBar.ts (hunk 1 above)
Comment (avnerbarr) [reply src/statusBar/pollStatusBar.ts:19]: state poller is state of user

status poller is status bar messages from the binary

File: src/registration/forceRegistration.ts (hunk 2 above)
Comment (dimacodota) [reply src/registration/forceRegistration.ts:129]: it is a more pragmatic way  of managing an async flow, as we did here :
https://github.com/codota/tabnine-vscode/blob/168deb7698863341024bd49322cc2ff915f8341c/src/capabilities/capabilities.ts#L62

Comment (avnerbarr) [reply src/registration/forceRegistration.ts:129]: what is the syntax for that?

File: src/statusBar/pollStatusBar.ts (hunk 1 above)
Comment (dimacodota) [reply src/statusBar/pollStatusBar.ts:19]: ok

File: src/registration/forceRegistration.ts (hunk 2 above)
Comment (avnerbarr) [reply src/registration/forceRegistration.ts:129]: this is just a completion handler. Where is it being resolved between 2 states

Comment (avnerbarr) [reply src/registration/forceRegistration.ts:129]: @dimacodota

**Review by yanir-codota — APPROVED** [review]

**Pull request merged**